src/app/shared/ui/menu/menu.component.ts

```diff
@@ -0,0 +1,71 @@
+import { ChangeDetectionStrategy, Component, inject, Input } from '@angular/core';
+import { MatListModule } from '@angular/material/list';
+import { MatExpansionModule } from '@angular/material/expansion';
+import { MatSlideToggleModule } from '@angular/material/slide-toggle';
+import { MatIconModule } from '@angular/material/icon';
+import { AsyncPipe, NgClass, NgFor, NgIf } from '@angular/common';
+import { NavigationEnd, Router, RouterModule } from '@angular/router';
+import { filter, map } from 'rxjs';
+import { CookieService } from 'ngx-cookie-service';
+
+import { MenuService } from './menu.service';
+import { menuItems } from './menu-items';
+import { UserRole } from '@core/store/user.interfaces';
+import { UserInitialsButtonComponent } from '../user-initials-button/user-initials-button.component';
+import { UserMenuComponent, UserMenuInputs } from '../user-menu';
+
+export interface MenuInputs {
+  role: UserRole;
+  userData: Required<UserMenuInputs>;
```

Comment:
O, to required ciekawe, jak to działa?

Reply:
Sprawia, że wszystkie pola interfejsu stają się `required` ;) ma to tutaj zastosowanie  bo interface `UserMenuInputs` wykorzystwany jest w innym miejscu i wszystkie pola są zapisane jako `optional`. 
Jakbyśmy chcieli zrobić to samo, ale w drugą stronę wtedy można użyć `Partial<>` 🙂